pkg/controller/atlasproject/alert_configurations.go

```diff
@@ -317,3 +337,35 @@ func isAlertConfigSpecEqualToAtlas(logger *zap.SugaredLogger, alertConfigSpec md
 
 	return true
 }
+
+func canAlertConfigurationReconcile(ctx context.Context, atlasClient mongodbatlas.Client, protected bool, project *mdbv1.AtlasProject, logger *zap.SugaredLogger) (bool, error) {
+	if !protected {
+		return true, nil
+	}
+
+	latestConfig := &mdbv1.AtlasProjectSpec{}
+	latestConfigString, ok := project.Annotations[customresource.AnnotationLastAppliedConfiguration]
+	if ok {
+		if err := json.Unmarshal([]byte(latestConfigString), latestConfig); err != nil {
+			return false, err
+		}
+	}
+
+	list, _, err := atlasClient.AlertConfigurations.List(ctx, project.ID(), &mongodbatlas.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	if len(list) == 0 {
+		return true, nil
+	}
+
+	diff := sortAlertConfigs(logger, latestConfig.AlertConfigurations, list)
+	if len(diff.Create) == 0 && len(diff.Delete) == 0 {
+		return true, nil
+	}
+
+	diff = sortAlertConfigs(logger, project.Spec.AlertConfigurations, list)
+
+	return len(diff.Create) == 0 || len(diff.Delete) == 0, nil
```

Comment:
I think we want both creations AND deletions to not be there to assume control of the alert configurations, don't we?

```suggestion
	return len(diff.Create) == 0 && len(diff.Delete) == 0, nil
```

Or, alternatively:
```suggestion
	return len(diff.Create) != 0 || len(diff.Delete) != 0, nil
```

Reply:
Otherwise, this is tricky enough to place a comment explaining the logic

Reply:
Good catch! I'll fix it